### DISCUSSION TITLE
ilogtail源码配置参数"polling_ignore_file_modify_timeout":7200的含义

### DISCUSSION THREAD

**Discussion by yecun123** [alibaba/ilogtail#679] [Help] [answered]
之前遇到一个问题，A日志文件正常被ilogtail日志文件采集，ilogtail因为不可控原因宕机，此时A日志文件还不断更新，切割生成新的历史文件B、C、D。 如果再次重启ilogtail 会发现只采集不断更新的日志文件A和之前ilogtail停止时的历史文件B。C和D被忽略掉了。 
添加 polling_ignore_file_modify_timeout 参数可以解决上述问题，但这个参数没明白啥意思，还有它对第一次采集无任何影响。 麻烦大佬们解释下，谢谢

**Comment by yyuuttaaoo** [comment, marked as answer]
超过这个polling_ignore_file_modify_timeout就认为是老的历史数据文件了，如果没有更新不会采集。B会采集是因为在checkpoint的中的文件会保证采完。

DEFINE_FLAG_INT32(ignore_file_modify_timeout, "if file modify time is up to XXX seconds, ignore it", 180);

    // Check the LMD of file, If the file is too old (180s), it will be ignored at
    // this round and wait for next round. Otherwise, a MODIFY event will be created
    // and pushed to @eventVec.

**Reply by yecun123** [reply]
感谢，这个参数我们自己测了很久，追问下，第一次安装filebeat，对历史日志文件判断有影响吗？（Filebeat下载解压后修改该参数，是否对历史日志文件的判断有影响）？
